onnxruntime/core/optimizer/transformer_memcpy.cc

```diff
@@ -255,14 +255,8 @@ void TransformerMemcpyImpl::ProcessDefs(onnxruntime::Node& node, const KernelReg
         provider_output_defs_.insert(arg);
     }
   } else if (node_provider_type != kCudaExecutionProvider && node_provider_type != kTensorrtExecutionProvider &&
-             node_provider_type != kRocmExecutionProvider && node_provider_type != kMIGraphXExecutionProvider) {
-    // TODO: copy between devices? i.e. multiple GPUs
-    if (node_provider_type != onnxruntime::kCpuExecutionProvider &&
-        node_provider_type != onnxruntime::kVitisAIExecutionProvider &&
-        !node_provider_type.empty()) {
-      ORT_THROW("Execution type '", node_provider_type, "' doesn't support memcpy ");
-    }
-
+             node_provider_type != kRocmExecutionProvider && node_provider_type != kMIGraphXExecutionProvider &&
+             node_provider_type != kWebGpuExecutionProvider) {
```

Comment:
```suggestion
             node_provider_type != kRocmExecutionProvider && node_provider_type != kMIGraphXExecutionProvider) {
```

Probably safer not to add here. We don't have DML or CANN so possibly having the CUDA/ROCm pairs to due to the special-casing of them in the `if` so that CUDA/TRT and ROCm/MIGraphX are treated as the same provider.